src/frequenz/sdk/actor/power_distributing/_component_status.py

```diff
@@ -72,3 +75,35 @@ class SetPowerResult:
 
     failed: typing.Iterable[int]
     """Component IDs for which the last set power command failed."""
+
+
+class ComponentStatusTracker(ABC):
+    """Interface for specialized component status trackers to implement."""
+
+    @abstractmethod
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        component_id: int,
+        max_data_age_sec: float,
+        max_blocking_duration_sec: float,
+        status_sender: Sender[ComponentStatus],
+        set_power_result_receiver: Receiver[SetPowerResult],
+    ) -> None:
+        """Create class instance.
+
+        Args:
+            component_id: Id of this battery
```

Comment:
I think here it should be `component` instead of `battery`

Reply:
true, will fix.

Reply:
https://github.com/frequenz-floss/frequenz-sdk-python/commit/61765a81